src/modules/UI/scenes/Login/Login.ui.js

```diff
@@ -20,8 +20,10 @@ class Login extends Component {
 
   componentWillReceiveProps (nextProps) {
     // If we have logged out, destroy and recreate the login screen:
-    if (this.props.account && nextProps.account !== this.props.account)
-      this.setState({key: this.state.key + 1})
+    if (this.props.account && (nextProps.account !== this.props.account))
+      if (typeof nextProps.account.username === 'undefined') {
+        this.setState({key: this.state.key + 1})
+      }
   }
```

Comment:
This would be simpler:

```js
if (
  this.props.account &&
  nextProps.account !== this.props.account &&
  nextProps.account.username === undefined
) {
  this.setState({ key: this.state.key + 1 })
}
```

Using `typeof x.y === 'undefined'` is unnecessary complication. Just do `x.y === undefined`.